src/extent.c

```diff
@@ -1930,6 +1967,11 @@ extent_merge_impl(tsdn_t *tsdn, arena_t *arena,
 		return true;
 	}
 
+	if (extent_dumpable_get(a) || extent_dumpable_get(b)) {
```

Comment:
We actually still need the logic in a lot of places, because of the possibility of aligned allocations (which can cause splits even inside of already split regions). We also have the merging issue either way, since we could have a no-dump extent next to a dumpable extent by chance or because of growth patterns.

I think the way we'd have to do it is by making "undumpable" its own extent state, and giving it its own extents_t in the arena. This is nice conceptually in some ways (if retained extents are ones we try to avoid using, retained, nondumpable extents are the ones we *really* try to avoid using), but OTOH I don't like the idea of adding another set of stats with different semantics, and shifting the dumping functionality to the arena code.